cg/meta/workflow/mutant/metrics_parser/metrics_parser.py

```diff
@@ -0,0 +1,131 @@
+    SampleResults,
+)
+from cg.store.models import Case
+
+
+class MutantResultsHeaderData:
+    SAMPLE_NAME: str = "sample_name"
+    SELECTION: str = "selection"
+    REGION_CODE: str = "region_code"
+    TICKET: str = "ticket"
+    PCT_N_BASES: str = "pct_n_bases"
+    PCT_10X_COVERAGE: str = "pct_10x_coverage"
+    QC_PASS: str = "qc_pass"
+    LINEAGE: str = "lineage"
+    PANGOLIN_DATA_VERSION: str = "pangolin_data_version"
+    VOC: str = "voc"
+    MUTATIONS: str = "mutations"
+
+
+class MutantResultsHeaderRawData:
+    SAMPLE_NAME: str = "Sample"
+    SELECTION: str = "Selection"
+    REGION_CODE: str = "Region Code"
+    TICKET: str = "Ticket"
+    PCT_N_BASES: str = "%N_bases"
+    PCT_10X_COVERAGE: str = "%10X_coverage"
+    QC_PASS: str = "QC_pass"
+    LINEAGE: str = "Lineage"
+    PANGOLIN_DATA_VERSION: str = "Pangolin_data_version"
+    VOC: str = "VOC"
+    MUTATIONS: str = "Mutations"
+
+
+class MetricsParser:
+    KEY_MAPPING: dict[str, str] = {
+        MutantResultsHeaderRawData.SAMPLE_NAME: MutantResultsHeaderData.SAMPLE_NAME,
+        MutantResultsHeaderRawData.SELECTION: MutantResultsHeaderData.SELECTION,
+        MutantResultsHeaderRawData.REGION_CODE: MutantResultsHeaderData.REGION_CODE,
+        MutantResultsHeaderRawData.TICKET: MutantResultsHeaderData.TICKET,
+        MutantResultsHeaderRawData.PCT_N_BASES: MutantResultsHeaderData.PCT_N_BASES,
+        MutantResultsHeaderRawData.PCT_10X_COVERAGE: MutantResultsHeaderData.PCT_10X_COVERAGE,
+        MutantResultsHeaderRawData.QC_PASS: MutantResultsHeaderData.QC_PASS,
+        MutantResultsHeaderRawData.LINEAGE: MutantResultsHeaderData.LINEAGE,
+        MutantResultsHeaderRawData.PANGOLIN_DATA_VERSION: MutantResultsHeaderData.PANGOLIN_DATA_VERSION,
+        MutantResultsHeaderRawData.VOC: MutantResultsHeaderData.VOC,
+        MutantResultsHeaderRawData.MUTATIONS: MutantResultsHeaderData.MUTATIONS,
+    }
+
+    @classmethod
+    def parse_samples_results(cls, case: Case, results_file_path: Path) -> dict[str, SampleResults]:
+        """Takes a case object and a results_file_path and resturns dict[str, SampleResults] with sample.internal_id as keys."""
+
+        raw_results: list[dict[str, Any]] = cls._get_raw_results(
+            results_file_path=results_file_path
+        )
+
+        validated_results_list: list[SampleResults] = cls._get_validated_results_list(
+            raw_results=raw_results
+        )
+
+        samples_results: dict[str, SampleResults] = cls._get_samples_results(
+            case=case, results_list=validated_results_list
+        )
+
+        return samples_results
+
+    @classmethod
+    def _get_raw_results(cls, results_file_path: Path) -> list[dict[str, Any]]:
+        """Parses raw_results from the results file."""
+        try:
+            raw_results: list[dict[Any, Any]] = [
+                sample_results
+                for sample_results in read_csv(file_path=results_file_path, read_to_dict=True)
+            ]
```

Comment:
I believe that the `read_csv` method already returns a list of dicts
```suggestion
            raw_results: list[dict[Any, Any]] = read_csv(file_path=results_file_path, read_to_dict=True)
```